### ISSUE TITLE
Subdomains paths in Pages / other reports

### ISSUE THREAD

**Issue by brenc** [matomo-org/matomo#14104]
We just rolled out Matomo (which is awesome). We noticed the same thing as [this user](https://forum.matomo.org/t/sub-domains-in-reports-pages-enty-pages-exit-pages/19186):

> The problem: In the “pages” report (and also the “entry pages” and “exit pages” reports) all URLs are mixed up independent on which sub domain they are running. First this is confusing, and second it seems to create wrong data! All index pages of the root and the sub domains are listed together as page URL “/index”. When I open the page overlay, the index page of the sub domain with the highest number of pageviews is displayed. But the data shown left hand in the “main metrics” section is the sum of all sub domains (pageviews, visits etc.). When I change the page overlay’s URL manually to another sub domain’s index page, the “main metrics” data stays the same.

So for example, we have URLs like this:

`https://www.newgrounds.com/art/view/diives`

In Matomo > Behavior > Pages it shows art > /index which is actually this URL:

`https://diives.newgrounds.com/art/`

Which is incorrect and unexpected.

We're already running this in our tracking code:

`_paq.push(["setDocumentTitle", document.domain + "/" + document.title]);`

But that doesn't seem to apply to Pages and other reports.

I searched around and I don't think this user put in a ticket so I thought I'd create one.

**Comment by mattab** [comment]
Thanks for the report @brenc It is indeed problematic and not good enough as it is now, as it's confusing and not clear, and many users like you don't expect this behavior. The issue is that some users want the behavior of considering all alias URLs as pointing to the same site (and merging them together), while others (like you) prefer to keep them separate. 

Maybe this is something we could address in Matomo 4.x?

**Comment by tsteur** [comment]
Not needed for Matomo 4. This can be done any time later.

**Comment by breaked01** [comment]
Hello.
Have a same issue with mixing of page urls which become from all subdomains, but they are logically separated from "main" and other domains, and may have different pages. What I can do at this moment to separate page urls?  Maybe just add a "subdomain" to page url variable (like in Google Analytics) or something like this? I can write own code if it needed, but want to get some info about where to start with that problem.

**Comment by diosmosis** [comment]
Hi @breaked01 , if you mean separate them in the Actions report, there's not much you can do at present. The change isn't that big though, so I created a poc in this PR: https://github.com/matomo-org/matomo/pull/17203

If acceptable it might be able to go in the next release.

CC @tsteur

**Comment by tsteur** [comment]
@diosmosis not sure I fully understand the PR correct but thinking as it's done in archiving, might need to add a comment that this will cause maybe issues when looking at week, month, year, ... that some will have the host while some rows won't and it'll be all mixed maybe?

**Comment by diosmosis** [comment]
> not sure I fully understand the PR correct but thinking as it's done in archiving, might need to add a comment that this will cause maybe issues when looking at week, month, year, ... that some will have the host while some rows won't and it'll be all mixed maybe?

yeah, that would be an issue. the easiest way around this might be to make the setting unchangeable. otherwise, if we see both types when aggregating we can group by host in the one that includes host, then aggregate w/ the one w/ no host. there's no way to go from no-host to host though so that's the best we can do I think.

**Comment by tsteur** [comment]
OK @diosmosis I guess we would just need to mention this in the comment for the config and also when we explain this new feature in an FAQ.

**Comment by uschindler** [comment]
There is already #17203 which looks like a good setting in the config to enable subdomains in the configuration. Unfortunately it was closed. Still waiting for it to be merged.

**Comment by uschindler** [comment]
> OK @diosmosis I guess we would just need to mention this in the comment for the config and also when we explain this new feature in an FAQ.

I think this is fine like any other setting (like the number of entries in subtables). If you change those settings you have to invalidate all reports and recreate them. When you setup Matomo you do this quite often.

This setting looks like something that people won't set all the time. It would be done by people who have subdomains with meaningful names and different content. Those people only having "www.example.org" and "example.org" would never enable this.

Maybe theres a way to automatically strip the "www." during reporting like at other places of Matomo.

**Comment by 9joshua** [comment]
Several users have expressed a need to see the domain in reports. Most are tracking multiple subdomains of the same TLD. Often they create Custom Reports as a limited workaround.
